presto-main/src/main/java/io/prestosql/server/PagesResponseWriter.java

```diff
@@ -82,6 +97,10 @@ public void writeTo(List<SerializedPage> serializedPages,
         try {
             SliceOutput sliceOutput = new OutputStreamSliceOutput(output);
             sliceOutput.writeInt(SERIALIZED_PAGES_MAGIC);
+            sliceOutput.writeLong(
```

Comment:
Do we care about backward compatibility of the protocol?
With such implementation one will have to upgrade whole cluster at one go. Upgrading node-by node will not be possible.

Probably it is hard to workaround that now. But even if we make the backward incompatible change now, we can extend the protocol in such way, that if we want to add something here in the future, we can do that in a way which allows rolling upgrade. One option to achieve that would be to put an extensible structure (protobuf/json/thrift) in place of raw sequence of longs.

Not sure if that is worth of an effort though.

Reply:
> Upgrading node-by node will not be possible.

it was never supported. E.g. currently, all nodes must declare same presto version or are not considered part of the cluster.
If we want to support rolling upgrade, we would need to do much more than just declare the intent.
Like certain provisions on the protocol level, but same for block serialization. Define version compatibility matrix/ranges.
And tests.

Since we're talking about this... I think this is dead end road in general. I am strongly convinced that rolling upgrade should be handled
on per-cluster rather than per-node basis. This delivers same behavior in most environments (except for fixed-size on-prem bare-metal 
environments), with way lower complexity.

Reply:
> I am strongly convinced that rolling upgrade should be handled on per-cluster rather than per-node basis

I think I agree actually. The behaviour is not the same, but similar. The cost is a bit different - but should no matter in most of the cases. And the "bare-metal" situation is very problematic - not sure if we care though. And similarly to bare-metal situation, spinning up a HUGE replacement cluster may result in stockout in public cloud zone.